js/ui/codeView.js

```diff
@@ -453,12 +459,24 @@ var CodingSession = new Lang.Class({
     },
 
     _setEffectsEnabled: function(actor, enabled) {
-        let desaturate = actor.get_effect('codeview-desaturate');
-        if (desaturate) {
-            desaturate.enabled = enabled;
+        let effect = actor.get_effect('codeview-effect');
+        if (effect) {
+            effect.enabled = enabled;
         } else {
-            desaturate = new Shell.CodeViewEffect({ enabled: enabled });
-            actor.add_effect_with_name('codeview-desaturate', desaturate);
+            let appInfo = this._shellApp.get_app_info();
+            let shaderEffect = appInfo.get_string(_HACK_SHADER_DESKTOP_KEY);
+            let shaderProto;
+
+            // By default, all apps get Shell.CodeViewEffect
+            if (shaderEffect)
```

Comment:
Nitpick: I'd rather set the `shaderEffect` to the key that has the default effect (`let shaderEffect = appInfo.get_string(_HACK_SHADER_DESKTOP_KEY) || 'desaturate';`). But of course this works too.

Reply:
Good suggestion, fixed.